core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -779,7 +779,7 @@ public Command workerHeartbeat(long workerId, Map<String, Long> usedBytesOnTiers
       return new Command(CommandType.Register, new ArrayList<Long>());
     }
 
-    synchronized (worker) {
+    synchronized (worker.getWorkerAddress()) {
```

Comment:
Why is this changed?

Reply:
This is really hacky now, re-purposing the mWorkerAddress field as an alternate way to lock the WorkerInfo. How are developers supposed to know when to synchronize on worker vs when to synchronize on `worker.getWorkerAddress()`? And why the worker address? I don't think this is a reasonable way to implement the locking here